helpers/dom.js

```diff
@@ -159,6 +159,21 @@ export function getPreviousAncestorSibling(node, predicate = () => true) {
 	return null;
 }
 
+export function getFirstDescendant(node, predicate) {
```

Comment:
Could `querySelectorComposed(...)` (defined below) serve your purpose?

If not, I propose we name this `getComposedFirstDescendant` to be consistent with the other helper methods that walk the composed DOM. The two immediately above are not composed, but ones that are, are named accordingly.

Also, I might suggest we default the predicate as is done in a couple others above (i.e. `(node, predicate = () => true)`), just for consistency.

I think we need be very careful with these methods that recursively walk deeper into the sub-tree, which includes the use of `querySelectorComposed`. Walking up the tree too, but I am slightly less concerned. They could be detrimental to performance if used in certain scenarios. I am somewhat worried that consumers will not consider this.

Reply:
I can make it work with `querySelectorComposed`, thanks Dave. I'll close this PR.